### PR TITLE
add subsample function

### PR DESCRIPTION
**Pull request by RNAer**
@amnona @mortonjt , could any of you review this PR?

### CODE REVIEW

File: calour/transforming.py [hunk 1]
```diff
@@ -345,3 +351,50 @@ def method(x): return x + delta
...
+
+    """
+    if inplace:
+        newexp = exp
+    else:
+        newexp = deepcopy(exp)
+    if newexp.sparse:
+        newexp.sparse = False
+    # subsample_counts() require int as input
+    newexp.data = newexp.data.astype(int)
```
Comment (amnona) [inline calour/transforming.py]: maybe worth testing it is actually integers so if not, raise an error that it only works on non-normalized data?
alternatively, can check exp.exp_metadata['normalized'] (we set it if we run exp.normalize)

**Comment by RNAer** [comment]
I can check exp.exp_metadata['normalized'], but it looks like biom read all
tables as float (that's why I did the int conversion), what do u suggest?

**Comment by amnona** [comment]
i think worth checking  exp.exp_metadata['normalized'] (to see if exists and non zero) and in that case raise data has been normalized.
Alternatively, can test if all (float) values are integers (i.e. x.0), but i think less good than checking 'normalized' since can imagine crazy case where after normalization all numbers are integers by chance :)
if we convert to int, maybe we should convert back to float after subsampling? (just to make sure biom table is always float)? or maybe not needed?

File: calour/transforming.py [hunk 2]
```diff
@@ -345,3 +351,50 @@ def method(x): return x + delta
...
+    drops = []
+    for row in range(newexp.data.shape[0]):
+        try:
+            newexp.data[row, :] = subsample_counts(newexp.data[row, :], n=total, replace=replace)
+        except ValueError:
+            drops.append(row)
```
Comment (mortonjt) [inline calour/transforming.py]: Couple of minor questions
1. What's the try for?
2. Can we just run an `apply_along_axis` function?

Comment (RNAer) [inline calour/transforming.py]: if replace is False, and a sample have smaller count than the total, this sample should be dropped. that's the `try` for. That's also why I am not sure `apply_along_axis` works in this situation.

**Comment by RNAer** [comment]
@amnona @mortonjt , updated with normalization check and more explaining documentation for the code

File: calour/transforming.py (hunk 2 above)
Comment (mortonjt) [inline calour/transforming.py]: 👍 ok that makes sense.  `apply_along_axis` probably won't work then.  Do you want to add a comment about why this occurs for future reference?

**Comment by RNAer** [comment]
yep. I did

**Pull request merged**